src/openms_gui/include/OpenMS/VISUAL/DIALOGS/TheoreticalSpectrumGenerationDialog.h

```diff
@@ -60,6 +60,47 @@ namespace OpenMS
 
 public:
 
+    // Note: If an additional checkbox is added all of the following three objects have to be edited!
+    // 
+    // 
+    // enum to get ion checkbox index (Ordering has to be the same as in the ui!)
+    enum class Checkbox
```

Comment:
I like this approach, but there is no need to expose it in the header.
This can all be moved into the cpp to reduce dependencies (vector etc)